kotlin/src/test/kotlin/io/logto/sdk/core/util/GenerateUtilsTest.kt

```diff
@@ -0,0 +1,43 @@
+package io.logto.sdk.core.util
+
+import com.google.common.truth.Truth.assertThat
+import org.junit.Test
+
+class GenerateUtilsTest {
+    @Test
+    fun generateCodeVerifierShouldBeRandomString() {
+        val code1 = GenerateUtils.generateCodeVerifier()
+        val code2 = GenerateUtils.generateCodeVerifier()
+        assertThat(code1).isNotEqualTo(code2)
+    }
+
+    @Test
+    fun generateCodeChallengeShouldBeDifferentWithDifferentCodeVerifiers() {
+        val code1 = GenerateUtils.generateCodeVerifier()
+        val codeChallenge1 = GenerateUtils.generateCodeChallenge(code1)
+
+        val code2 = GenerateUtils.generateCodeVerifier()
+        val codeChallenge2 = GenerateUtils.generateCodeChallenge(code2)
+
+        assertThat(codeChallenge1).isNotEqualTo(codeChallenge2)
+    }
```

Comment:
we should also assert static return value for code challenge, see [this file](https://github.com/logto-io/swift/blob/master/Tests/LogtoTests/LogtoUtilitiesTests.swift#L18)